tests/linalg_test.py

```diff
@@ -1373,6 +1373,82 @@ def expm(x):
+            "testcase_name":
+            "_shape={}".format(jtu.format_shape_dtype_string(shape, dtype)),
+            "shape": shape, "dtype": dtype
+        } for shape in [(4, 4), (15, 15), (50, 50), (100, 100)]
+                            for dtype in float_types + complex_types))
+  @jtu.skip_on_devices("gpu", "tpu")
+  def testSchur(self, shape, dtype):
+      rng = jtu.rand_default(self.rng())
+      args_maker = lambda: [rng(shape, dtype)]
+
+      self._CheckAgainstNumpy(osp.linalg.schur, jsp.linalg.schur, args_maker)
+      self._CompileAndCheck(jsp.linalg.schur, args_maker)
+
+  @parameterized.named_parameters(
+    jtu.cases_from_list({
+        "testcase_name":
+        "_shape={}".format(jtu.format_shape_dtype_string(shape, dtype)),
+        "shape" : shape, "dtype" : dtype
+    } for shape in [(4, 4), (15, 15), (50, 50), (100, 100)]
+      for dtype in float_types + complex_types))
+  @jtu.skip_on_devices("gpu", "tpu")
+  def testSqrtmPSDMatrix(self, shape, dtype):
+    # Checks against scipy.linalg.sqrtm when the principal square root
+    # is guaranteed to be unique (i.e no negative real eigenvalue)
+    rng = jtu.rand_default(self.rng())
+    arg = rng(shape, dtype)
+    mat = arg @ arg.T
+    args_maker = lambda : [mat]
+    if dtype == np.float32 or dtype == np.complex64:
+        tol = 1e-4
+    else:
+        tol = 1e-8
+    self._CheckAgainstNumpy(osp.linalg.sqrtm,
+                            jsp.linalg.sqrtm,
+                            args_maker,
+                            tol=tol,
+                            check_dtypes=False)
+    self._CompileAndCheck(jsp.linalg.sqrtm, args_maker)
+
+  @parameterized.named_parameters(
+  jtu.cases_from_list({
+      "testcase_name":
+      "_shape={}".format(jtu.format_shape_dtype_string(shape, dtype)),
+      "shape" : shape, "dtype" : dtype
+  } for shape in [(4, 4), (15, 15), (50, 50), (100, 100)]
+    for dtype in float_types + complex_types))
+  @jtu.skip_on_devices("gpu", "tpu")
+  def testSqrtmGenMatrix(self, shape, dtype):
+    rng = jtu.rand_default(self.rng())
+    arg = rng(shape, dtype)
+    if dtype == np.float32 or dtype == np.complex64:
+      tol = 1e-3
+    else:
+      tol = 1e-8
+    R = jsp.linalg.sqrtm(arg)
+    self.assertAllClose(R @ R, arg, atol=tol, check_dtypes=False)
+
+  @parameterized.named_parameters(
+  jtu.cases_from_list({
+      "testcase_name":
+      "_shape={}".format(jtu.format_shape_dtype_string(shape, dtype)),
+      "shape" : shape, "dtype" : dtype
+  } for shape in [(4, 4), (15, 15), (50, 50), (100, 100)]
+    for dtype in float_types + complex_types))
+  @jtu.skip_on_devices("gpu", "tpu")
+  def testSqrtmBatching(self, shape, dtype):
+    rng = jtu.rand_default(self.rng())
+    batch_size = 10
+    shape = (batch_size, ) + shape
+    args = rng(shape, dtype)
+    square = vmap(lambda mat: mat @ mat)
+
+    roots = vmap(jsp.linalg.sqrtm)(args)
+    self.assertAllClose(square(roots), args, atol=5e-3, check_dtypes=False)
```

Comment:
I don't think we really need this test for the current implementation of `sqrtm`, because it's built out of other batchable JAX primitives.

(It's different for the tests that are checking linear algebra routines with custom batching rules.)